### PR TITLE
PpmSeq_qc_update_h5_format

### PR DESCRIPTION
**Pull request by kenissur**

### CODE REVIEW

File: ugvc/mrd/ppmSeq_utils.py [hunk 1]
```diff
@@ -773,17 +775,19 @@ def read_trimmer_tags_dataframe(
         df_tags = df_strand_ratio_category.drop(ppmSeqCategories.END_UNREACHED.value, errors="ignore")[
             [HistogramColumnNames.STRAND_RATIO_CATEGORY_START.value]
         ]
-        df_tags.index = [f"% {x} reads" for x in df_tags.index]
+        df_tags.index = [f"PCT_{x}_reads" for x in df_tags.index]
```
Comment (inbalzelig-ugi) [inline ugvc/mrd/ppmSeq_utils.py:782]: better pct_ then PCT_ but it doesn't really matter

File: ugvc/mrd/ppmSeq_utils.py [hunk 2]
```diff
@@ -1049,7 +1052,7 @@ def collect_statistics(
         adapter_version=adapter_version,
         df_strand_ratio_category=df_strand_ratio_category,
         df_category_consensus=df_category_consensus,
-        df_sorter_stats=df_sorter_stats,
+        df_sorter_stats=df_sorter_stats.to_frame(),
```
Comment (inbalzelig-ugi) [inline ugvc/mrd/ppmSeq_utils.py]: df_sorter_stats in line 1048 is not a df, but a series. Consider renaming it

File: ugvc/mrd/ppmSeq_utils.py [hunk 3]
```diff
@@ -950,7 +953,7 @@ def read_dumbell_leftover_from_trimmer_histogram(trimmer_histogram_extra_csv, le
         dataframe with dumbell leftover stats
     """
     df_dumbbell_leftover = pd.read_csv(trimmer_histogram_extra_csv)
-    expected_columns = ["Dumbbell_leftover_start_match", "count"]
+    expected_columns = ["dumbbell_leftover_start_match", "count"]
```
Comment (inbalzelig-ugi) [inline ugvc/mrd/ppmSeq_utils.py]: consider moving the string "dumbbell_leftover_start_match" to a const or enum if you use it in multiple places

File: ugvc/mrd/ppmSeq_utils.py [hunk 4]
```diff
@@ -1071,26 +1074,40 @@ def collect_statistics(
...
+            store["strand_ratio_category_concordance"] = df_category_concordance
+            store["strand_ratio_category_consensus"] = df_category_consensus
+            keys_to_convert += [
+                "strand_ratio_category_concordance",
+                "strand_ratio_category_consensus",
+            ]
         if trimmer_failure_codes_csv:
-            store["df_trimmer_failure_codes"] = df_trimmer_failure_codes
+            store["trimmer_failure_codes"] = df_trimmer_failure_codes
+            keys_to_convert += ["trimmer_failure_codes"]
```
Comment (inbalzelig-ugi) [inline ugvc/mrd/ppmSeq_utils.py:1111]: I wonder if you need trimmer_failure_codes in your h5... When running on-tool, these should be uploaded anyway. Same for sorter stats (unless you are making changes)

File: ugvc/mrd/ppmSeq_utils.py [hunk 5]
```diff
@@ -1397,7 +1414,7 @@ def plot_strand_ratio_category_concordnace(
         ("All reads", "Only reads where end was reached"),
         (df_category_concordance, df_category_concordance_no_end_unreached),
     ):
-        df_plot = df_plot.unstack().droplevel(0, axis=1)
+        df_plot = df_plot.to_frame().unstack().droplevel(0, axis=1)
```
Comment (inbalzelig-ugi) [inline ugvc/mrd/ppmSeq_utils.py:1422]: Again- if you need to_frmae() then it is not a df. Consider renaming to be more clear

File: ugvc/mrd/ppmSeq_utils.py [hunk 6]
```diff
@@ -1734,14 +1751,18 @@ def convert_h5_to_papyrus_json(h5_file: str, output_json: str) -> str:
...
     flatten_dfs.update({key: flatten_metrics(h5_file, key) for key in keys_to_convert if f"/{key}" in h5_file_keys})
 
     # category concordance
-    if "/df_category_concordance" in h5_file_keys:
-        df_category_concordance = pd.read_hdf(h5_file, "/df_category_concordance")
+    if "/strand_ratio_category_concordance" in h5_file_keys or "strand_ratio_category_concordance" in h5_file_keys:
+        df_category_concordance = pd.read_hdf(h5_file, "/strand_ratio_category_concordance")
         df_category_concordance.index = df_category_concordance.index.to_flat_index()
-        df_category_concordance.rename(columns={"count": 0}, inplace=True)
+        df_category_concordance = df_category_concordance.to_frame().rename(columns={"count_norm": 0})
```
Comment (inbalzelig-ugi) [inline ugvc/mrd/ppmSeq_utils.py]:
```suggestion
        df_category_concordance = df_category_concordance.to_frame().rename(columns={HistogramColumnNames.COUNT_NORM.value: 0})
```

File: ugvc/utils/metrics_utils.py [hunk 7]
```diff
@@ -188,8 +189,8 @@ def read_sorter_statistics_csv(sorter_stats_csv: str, edit_metric_names: bool =
 
     if edit_metric_names:
         # rename metrics to uniform convention
-        df_sorter_stats = df_sorter_stats.rename({c: c.replace("PCT_", "% ") for c in df_sorter_stats.index})
-    return df_sorter_stats
+        df_sorter_stats = df_sorter_stats.rename({c: c.replace("% ", "PCT_") for c in df_sorter_stats.index})
```
Comment (inbalzelig-ugi) [inline ugvc/utils/metrics_utils.py:192]:
```suggestion
        df_sorter_stats = df_sorter_stats.rename({c: c.replace("% ", "pct_") for c in df_sorter_stats.index})
```

**Review by inbalzelig-ugi — APPROVED** [review]
LGTM.
Few comments about the naming convention, you can choose to implement it or not. I'm approving the PR anyway

File: ugvc/mrd/ppmSeq_utils.py (hunk 5 above)
Comment (kenissur) [reply ugvc/mrd/ppmSeq_utils.py:1422]: it is after I convert and unstack it

File: ugvc/mrd/ppmSeq_utils.py (hunk 3 above)
Comment (kenissur) [reply ugvc/mrd/ppmSeq_utils.py]: Done

File: ugvc/mrd/ppmSeq_utils.py (hunk 1 above)
Comment (kenissur) [reply ugvc/mrd/ppmSeq_utils.py:782]: I chose to be consistent with the Sorter metrics because I present them in the same table as the ppmSeq stats

**Pull request merged**